### PR TITLE
[Feature] support parquet scanner to split task

### PR DESCRIPTION
**Pull request by dirtysalt**
## Why I'm doing:

Since connector framework supports split morsel features, we want parquet scanner to support it.

## What I'm doing:

To support parquet scanner to split tasks.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/formats/parquet/file_reader.cpp [hunk 1]
```diff
@@ -190,6 +212,34 @@ Status FileReader::_get_footer() {
...
+        int64_t end_offset = _get_row_group_end_offset(row_group);
+        auto split_ctx = std::make_unique<SplitContext>();
+        split_ctx->split_start = start_offset;
+        split_ctx->split_end = end_offset;
+        split_ctx->file_metadata = _file_metadata;
+        _scanner_ctx->split_tasks->emplace_back(std::move(split_ctx));
+    }
+    // if only one split task, clear it, no need to do split work.
+    if (_scanner_ctx->split_tasks->size() <= 1) {
+        _scanner_ctx->split_tasks->clear();
```
Comment (Youngwb) [inline be/src/formats/parquet/file_reader.cpp:236]: add split_context,  set split_start and split_end?

File: be/src/formats/parquet/file_reader.cpp [hunk 2]
```diff
@@ -190,6 +212,34 @@ Status FileReader::_get_footer() {
     return Status::OK();
 }
 
+void FileReader::_build_split_tasks() {
+    if (!_scanner_ctx->enable_split_tasks || _scanner_ctx->split_context != nullptr ||
```
Comment (Youngwb) [inline be/src/formats/parquet/file_reader.cpp:216]: It's better to add some comment to explain split_context and split_tasks, it's a little confusing

File: be/src/formats/parquet/file_reader.cpp [hunk 3]
```diff
@@ -190,6 +212,34 @@ Status FileReader::_get_footer() {
...
+        const tparquet::RowGroup& row_group = _file_metadata->t_metadata().row_groups[i];
+        bool selected = _select_row_group(row_group);
+        if (!selected) continue;
+        int64_t start_offset = _get_row_group_start_offset(row_group);
+        int64_t end_offset = _get_row_group_end_offset(row_group);
+        auto split_ctx = std::make_unique<SplitContext>();
+        split_ctx->split_start = start_offset;
+        split_ctx->split_end = end_offset;
+        split_ctx->file_metadata = _file_metadata;
+        _scanner_ctx->split_tasks->emplace_back(std::move(split_ctx));
```
[the remark this answers is not in the document]
Comment (dirtysalt) [reply be/src/formats/parquet/file_reader.cpp:232]: will do merge in next pr.

File: be/src/formats/parquet/file_reader.cpp (hunk 1 above)
Comment (dirtysalt) [reply be/src/formats/parquet/file_reader.cpp:236]: nono, that's not how it works. 

Here we just do split, connector frame work will set right split context for us in next round.

File: be/src/formats/parquet/file_reader.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/formats/parquet/file_reader.cpp:216]: OK

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergify backport branch-3.2

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Pull request merged**